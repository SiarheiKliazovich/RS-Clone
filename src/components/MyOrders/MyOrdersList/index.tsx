import { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";

import { IOrdersListComponent } from "../../../interfaces/ordersListComponent";

const MyOrdersList: FunctionComponent<IOrdersListComponent> = ({ orders }) => {
  const { t } = useTranslation("dataLang");

  return (
    <div className="market-block__products">
      <div className="row__products">
        {orders.map((item, i) => {
          return (
            <div key={i} className="tie__product my-ties">
              <div className="tie__products_img">
                <div>
                  <img className="products_img" src={item.image} alt="" />
                </div>
              </div>
              <div className="tie__products_discription my-orders-list__item">
                <p className="tie__products_name">{t("myOrders.data")}</p>
                <div className="item__date">
                  {t("myOrders.date")}: {item.date.slice(0, 10)}
                </div>
                <div className="item__price">
                  {t("myOrders.sum")}: {item.price}$
                </div>
                <div className="item__status">
                  {t("myOrders.status")}: {item.status}
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default MyOrdersList;
