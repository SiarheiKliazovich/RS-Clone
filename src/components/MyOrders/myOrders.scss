@import "../../sass/variables";
@import "../../sass/mixins";

.my-orders__wrapper > {
    box-sizing: border-box;
}

.my-orders__wrapper {
    width: 1000px;
    margin: 120px auto 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: $color-element-background-light;
    padding: 40px 0 10px 5px;
    border-radius: 5px;
    box-shadow: $box-shadow;

    @include media-tablet {
        width: 900px;
    }

    @include media-small-tablet {
        width: 700px;
    }

    @include media-mobile {
        padding-top: 40px;
        width: 500px;
    }

    @include media-small-mobile {
        width: 340px;
    }
}

.my-orders__header {
    margin-top: -70px;
}

.my-orders-list__wrapper {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    margin: 10px auto 0;
    gap: 20px;
}

.my-orders-list__item {
    width: 300px;
    display: flex;
    justify-content: space-between;
    border: 1px solid $font-color-dark;
    overflow: hidden;
    border-radius: 5px;
    margin-bottom: 20px;
    height: 180px;    

    h4 {
        margin-bottom: 10px;
    }

    .item__image {
        img {
            width: 130px;
        }
    }
    
    .item__text {
        width: 165px;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        margin: 0 5px;
    }
    
    .item__date,
    .item__price,
    .item__status {
        margin-bottom: 10px;
    }
    
    .item__button {
        width: 120px;
        cursor: pointer;
        border-radius: 5px;
        border: none;
        padding: 3px;
    }
}
