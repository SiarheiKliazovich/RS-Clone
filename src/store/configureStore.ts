/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
