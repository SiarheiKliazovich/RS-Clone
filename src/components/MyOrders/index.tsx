import { FunctionComponent, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";

import {
  myOrdersFetching,
  myOrdersFetched,
  myOrdersFetchingError,
} from "../../actions";
import tieMarketLang from "../../data/tieMarket";
import { ILangReducer } from "../../interfaces/langReducer";
import { lightTheme, nightTheme } from "../../data/constants";
import { IStore } from "../../interfaces/store";
import { IAuthReducer } from "../../interfaces/authReducer";
import { getOrdersByUserId } from "../../services/apiOrders";
import Spinner from "../Spinner";
import { IMyOrdersReducer } from "../../interfaces/myOrdersReducer";
import ErrorMessage from "../ErrorMessage";
import MyOrdersList from "./MyOrdersList";

import "./myOrders.scss";

const MyOrders: FunctionComponent = () => {
  const interfaceSettings = useSelector((state: IStore) => state.appInterface);
  const { isNightMode } = interfaceSettings;
  const backgroundColor = isNightMode
    ? nightTheme.background.element
    : lightTheme.background.element;

  const { lang } = useSelector((state: ILangReducer) => state.langReducer);
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const list = tieMarketLang.find((c) => c.lang === lang)!;

  const authStore = useSelector((state: IAuthReducer) => state.auth);
  const { user } = authStore;

  const myOrdersStore = useSelector(
    (state: IMyOrdersReducer) => state.myOrdersReducer
  );
  const { ordersLoadingStatus, orders } = myOrdersStore;

  const dispatch = useDispatch();

  const getOrders = async (): Promise<void> => {
    try {
      dispatch(myOrdersFetching());
      const orders = await getOrdersByUserId(user._id);
      dispatch(myOrdersFetched(orders));
    } catch {
      dispatch(myOrdersFetchingError());
    }
  };

  useEffect(() => {
    getOrders();
  }, []);

  const spinner = ordersLoadingStatus === "loading" ? <Spinner /> : null;

  return (
    <div className="tiemarket-wrapper">
      <div
        className="tiemarket__banner-wrapper"
        style={{
          backgroundColor: isNightMode
            ? nightTheme.background.page
            : lightTheme.background.page,
        }}
      >
        <div className="container container--high">
          <div className="market-block" style={{ backgroundColor }}>
            <div className="market-block__title_wrapper">
              <h4
                className="market-block__title"
                style={{
                  color: isNightMode
                    ? nightTheme.fontColor
                    : lightTheme.fontColor,
                }}
              >
                {list.data.myorders}
              </h4>
            </div>
            {spinner}
            {ordersLoadingStatus === "idle" ? (
              <MyOrdersList orders={orders} />
            ) : null}
            {ordersLoadingStatus === "error" ? <ErrorMessage /> : null}
          </div>
        </div>
      </div>
    </div>
  );
};

export default MyOrders;
