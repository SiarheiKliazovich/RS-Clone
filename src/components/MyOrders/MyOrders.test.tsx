import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import MyOrders from ".";

const mockStore = configureStore([]);

describe("MyOrders", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
      langReducer: {
        lang: "en",
      },
      auth: {
        user: {
          role: "USER",
        },
      },
      myOrdersReducer: {
        ordersLoadingStatus: "idle",
        orders: [],
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <MyOrders />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
