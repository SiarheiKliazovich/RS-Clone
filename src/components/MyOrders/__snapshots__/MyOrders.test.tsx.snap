// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrders renders correctly 1`] = `
<div
  class="tiemarket-wrapper"
  style="background-color: rgb(255, 255, 255);"
>
  <div
    class="tiemarket__banner-wrapper"
  >
    <div
      class="container"
    >
      <div
        class="market-block"
      >
        <div
          class="market-block__title_wrapper"
        >
          <h4
            class="market-block__title"
          >
            My Orders
          </h4>
        </div>
        <div
          class="market-block__products"
        >
          <div
            class="row__products"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
